### PR TITLE
Explicitly set lvm device source

### PR DESCRIPTION
**Pull request by schaefi**
Set external_device_info_source=none for pvcreate call
This is related to Issue #1665

### CODE REVIEW

**Comment by schaefi** [comment]
@seife ok let's put this in code and see if it fixes the lvm build issues when no udev is present

**Comment by seife** [comment]
I had to add the external_device_info_source=none to *all* LVM commands, otherwise it would error out just a bit later ;-)

This is the patch I'm applying locally:
```
Index: b/kiwi/volume_manager/lvm.py
===================================================================
--- a/kiwi/volume_manager/lvm.py
+++ b/kiwi/volume_manager/lvm.py
@@ -108,8 +108,9 @@ class VolumeManagerLVM(VolumeManagerBase
             command=['vgremove', '--force', volume_group_name],
             raise_on_error=False
         )
-        Command.run(['pvcreate', self.device])
-        Command.run(['vgcreate', volume_group_name, self.device])
+
+        Command.run(['pvcreate', '--config', 'devices/external_device_info_source=none', self.device])
+        Command.run(['vgcreate', '--config', 'devices/external_device_info_source=none', volume_group_name, self.device])
         self.volume_group = volume_group_name
 
     @staticmethod
@@ -135,13 +136,13 @@ class VolumeManagerLVM(VolumeManagerBase
             '--> running "lvcreate -Zn %s"', " ".join(lvcreate_args)
         )
         Command.run(
-            ['lvcreate', '-Zn'] + lvcreate_args
+            ['lvcreate', '--config', 'devices/external_device_info_source=none', '-Zn'] + lvcreate_args
         )
         log.debug(
             '--> running "vgscan --mknodes" to create missing nodes'
         )
         Command.run(
-            ['vgscan', '--mknodes']
+            ['vgscan', '--config', 'devices/external_device_info_source=none', '--mknodes']
         )
 
     def create_volumes(self, filesystem_name):
@@ -359,7 +360,7 @@ class VolumeManagerLVM(VolumeManagerBase
             if self.umount_volumes():
                 Path.wipe(self.mountpoint)
                 try:
-                    Command.run(['vgchange', '-an', self.volume_group])
+                    Command.run(['vgchange', '--config', 'devices/external_device_info_source=none', '-an', self.volume_group])
                 except Exception:
                     log.warning(
                         'volume group %s still busy', self.volume_group
```

**Comment by schaefi** [comment]
done, let's see if it makes a difference

File: kiwi/volume_manager/lvm.py [hunk 1]
```diff
@@ -167,7 +178,7 @@ def create_volumes(self, filesystem_name):
                 '--> volume %s with %s MB', volume.name, volume_mbsize
             )
             self._create_volume_no_zero(
-                [
+                self.lvm_tool_options, [
                     '-L', format(volume_mbsize), '-n',
```
Comment (seife) [inline kiwi/volume_manager/lvm.py:182]: looks like a duplicate, _create_volume_no_zero already adds lvm_tool_options

File: kiwi/volume_manager/lvm.py [hunk 2]
```diff
@@ -188,7 +199,7 @@ def create_volumes(self, filesystem_name):
             full_size_volume = canonical_volume_list.full_size_volume
             log.info('--> fullsize volume %s', full_size_volume.name)
             self._create_volume_no_zero(
-                [
+                self.lvm_tool_options, [
```
Comment (seife) [inline kiwi/volume_manager/lvm.py:202]: same here

**Review by seife — CHANGES_REQUESTED** [review]

**Comment by schaefi** [comment]
For reference the integration test build to watch

* https://build.opensuse.org/package/show/Virtualization:Appliances:Images:Testing_x86:suse/test-image-lvm

File: kiwi/volume_manager/lvm.py (hunk 1 above)
Comment (seife) [reply kiwi/volume_manager/lvm.py:182]: sorry, disregard. did not notice that _create_volume_no_zero got an addtional option.

File: kiwi/volume_manager/lvm.py (hunk 2 above)
Comment (schaefi) [reply kiwi/volume_manager/lvm.py:202]:
```python
@staticmethod
    def _create_volume_no_zero(lvm_tool_options, lvcreate_args):
        ...
```

the private helper method gets self.lvm_tool_options as argument. I don't see an error ?

**Comment by schaefi** [comment]
I forgot vgcreate

**Comment by schaefi** [comment]
@seife hooray works :)

**Review by davidcassany — APPROVED** [review]
Looks good! I am glad we manage to work around the lvm issue. I am wondering if this will strike us back at some point in case udev is required for some specific scenario... but I guess we can actually wait and see, and meanwhile we have our tests functional :)

**Comment by seife** [comment]
It's already prepared to be configured / conditionalized if I read the code correctly, :-)

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**